style/swift.md

```diff
@@ -1,3 +1,1037 @@
+- preferring methods and properties to free functions
+- casing acronyms and initialisms uniformly up or down. Use `ID` when part of a type but `Id` when part of an identifier's name
+- giving the same base name to methods that share the same meaning
+- avoiding overloads on return type
+- choosing good parameter names that serve as documentation
+- preferring to name the first parameter instead of including its name in the method name, except as mentioned under [Delegates](#delegates)
+- labeling closure and tuple parameters
+- taking advantage of default parameters
+- prefer to locate parameters with defaults toward the end
+
+### Class Prefixes
+
+Swift types are automatically namespaced by the module that contains them and you should not add a class prefix. If two names from different modules collide you can disambiguate by prefixing the type name with the module name. However, only specify the module name when there is possibility for confusion, which should be rare.
+
+```swift
+import SomeModule
+
+let myClass = MyModule.UsefulClass()
+```
+
+### Delegates
+
+When creating custom delegate methods, an unnamed first parameter should be the delegate source. (UIKit contains numerous examples of this.)
+
+**Preferred**:
+```swift
+func namePickerView(_ namePickerView: NamePickerView, didSelectName name: String)
+func namePickerViewShouldReload(_ namePickerView: NamePickerView) -> Bool
+```
+
+**Not Preferred**:
+```swift
+func didSelectName(namePicker: NamePickerViewController, name: String)
+func namePickerShouldReload() -> Bool
+```
+
+### Use Type Inferred Context
+
+Use compiler inferred context to write shorter, clear code.  (Also see [Type Inference](#type-inference).)
+
+**Preferred**:
+```swift
+let selector = #selector(viewDidLoad)
+view.backgroundColor = .red
+let toView = context.view(forKey: .to)
+let view = UIView(frame: .zero)
+```
+
+**Not Preferred**:
+```swift
+let selector = #selector(ViewController.viewDidLoad)
+view.backgroundColor = UIColor.red
+let toView = context.view(forKey: UITransitionContextViewKey.to)
+let view = UIView(frame: CGRect.zero)
+```
+
+### Generics
+
+Generic type parameters should be descriptive, upper camel-case names. When a type name doesn't have a meaningful relationship or role, use a traditional single uppercase letter such as `T`, `U`, or `V`.
+
+**Preferred**:
+```swift
+struct Stack<Element> { ... }
+func write<Target: OutputStream>(to target: inout Target)
+func swap<T>(_ a: inout T, _ b: inout T)
+```
+
+**Not Preferred**:
+```swift
+struct Stack<T> { ... }
+func write<target: OutputStream>(to target: inout target)
+func swap<Thing>(_ a: inout Thing, _ b: inout Thing)
+```
+
+### Language
+
+Use US English spelling to match Apple's API.
+
+**Preferred**:
+```swift
+let color = "red"
+```
+
+**Not Preferred**:
+```swift
+let colour = "red"
+```
+
+## Code Organization
+
+Use extensions to organize your code into logical blocks of functionality. Add `MARK`s if they help with it.
```

Comment:
Not sure why you'd use extensions for this? Extensions have a functional purpose e.g. extending objects from third party frameworks or the standard framework which we don't have access to or prefer to not edit directly. It is also used for extending protocols to provide polymorphism without inheritance.

Using them for protocol adherence can be useful, and also for separating object code across files. But to use them to organize code when we already have `//MARK` (which is actually more useful in xcode) doesn't make sense and just adds noise.

Although I don't strongly object to using `extension` to organise code, I wouldn't go out of my way to promote it's usage in that way (it seems like a classic case of "when you're holding a hammer, everything looks like a nail").

Reply:
I agree with Daniel